backend/src/controllers/user.go

```diff
@@ -33,3 +34,33 @@ func (u *UserController) GetAllUsers(c *fiber.Ctx) error {
 
 	return c.Status(fiber.StatusOK).JSON(users)
 }
+
+// UpdateUser godoc
+//
+// @Summary		Updates a user
+// @Description	Updates a user
+// @ID			update-user-by-id
+// @Tags      	user
+// @Produce		json
+// @Success		200	  {object}	  models.User
+// @Failure     404   {string}    string "User not found"
+// @Failure 	400   {string}    string "Failed to update user"
```

Comment:
does this message fit the [definition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) of a 400 status code? once you update your transactions you will realize there will be changes to this godoc. also, as a convention we're going to use all lower for err messages (see [this](https://github.com/GenerateNU/sac/blob/main/backend/src/controllers/category.go#L18))